src/smoosh/src/smoosh_channel.erl

```diff
@@ -232,22 +232,24 @@ handle_info(unpause, State0) ->
     State = maybe_open_queue(State0),
     {noreply, maybe_start_compaction(State#state{paused = false})}.
 
-terminate(_Reason, #state{name = Name, waiting = Q}) ->
-    file:delete(active_file_name(Name)),
-    file:delete(starting_file_name(Name)),
-    if
-        Q =/= nil ->
-            smoosh_priority_queue:close(Q);
-        true ->
-            nil
-    end,
+terminate(_Reason, _State) ->
     ok.
 
 maybe_recover_state(#state{name = Name} = State) ->
     Active = recover(active_file_name(Name)),
     Starting = recover(starting_file_name(Name)),
-    DatabaseDir = config:get("couchdb", "database_dir"),
-    Active1 = get_matching_compact_files(DatabaseDir, Active),
+    Active1 = lists:foldl(
+        (fun({DbName, CPid}, Acc) ->
+            case couch_server:is_compacting(DbName) of
+                true ->
+                    [{DbName, CPid} | Acc];
```

Comment:
I don't think it would work. Because `CPid` wouldn't be available after node restart. We would have to enqueue the Db again.

Reply:
See related suggestion to track the NodeStartTime https://github.com/apache/couchdb/pull/3766#discussion_r774761365

Reply:
I thought about it over weekend and want to share one idea that you have.

Instead of distinguishing between VM restart and smoosh components restart we could try to use `is_process_alive` and use `{DbName, CPid}` in this case. To be extra sure we don't run into Pid recycle problem we could have a method which checks type of a pid.

Reply:
If we call `is_process_alive` for each `CPid`, I believe we will run into the problem where we desync our `active` in the `state`. For example, if we found a `CPid` that is not alive and call `enqueue` on the `DbName` for that `CPid`, it will enqueue and record the `DbName` in `active`. However, at the end of `maybe_recover_state` we overwrite the existing `active`: ```State#state{active = Active1, starting = Starting}```. Maybe it would be safer to use the timestamp approach you recommended but in an "all or nothing" approach where if not all `CPid`s are alive then we enqueue each DB that we recovered from the state file.

Reply:
I meant doing [this](https://github.com/apache/couchdb/pull/3766/files#diff-113cd247cfe56b1aa882a37daa2856fd33bab2e2a6183034eb3a30088960dc13R333) by enqueue.
However the fold loop would have to be restructured because we would be updating the waiting in the loop.